build-system/tasks/presubmit-checks.js

```diff
@@ -1474,6 +1499,8 @@ async function presubmit() {
 
 module.exports = {
   presubmit,
```

Comment:
**`amp presubmit-checks`** is unchanged, but it could ignore reporting `*.js` files that are linted. This would prevent dupe reports.

Reply:
An initial comment, until I have the time to review more fully: I like the general direction of this PR, and think we should fully move `presubmit.js` into a local `eslint` plugin and completely get rid of `amp presubmit`. Presubmit only takes a couple of seconds for all files, and I think it will be much more usable when done as part of `lint` because of IDE-level red-underlining.

Would it be possible to fully migrate `presubmit` into `lint` in this PR?

Reply:
Aaah. Non-JS files. Okay. Lemme review this properly later today.

Reply:
Yeah, sorry if was too vague 🙂

We might have to move a subset of these globs into eslint configuration to enable this rule.

https://github.com/ampproject/amphtml/blob/3f4851ffa5b9da1362774bb314e5c42a85c1921f/build-system/test-configs/config.js#L127-L158

Reply:
@jridgewell is adding a way to have the `prettier` IDE plugin ignore specific files by reading them from `.prettierignore` in #33449. I think we can (and should) reuse that functionality for `eslint` via `.eslintignore`. Added him as reviewer to get his take.

Also, if all this is too much for one PR, feel free to punt and I can chip in with a follow-up. (I sometimes get greedy when people voluntarily improve infrastructure code 😃 )

Reply:
I don't know if I understand: flip this so that `presubmitGlobs` consumes `.eslintignore`? 

My idea was to remove all responsibility for JS files from `presubmit`. Then, some of the presubmit ignore rules are transferred into `.eslintignore`. `eslint` excludes those already.

Reply:
Eslint already ignores with its `.eslintignore` file, we don't need any special code for it. Prettier, when using the API, does not respect the `.prettierignore`. (Prettier only ignores when using the CLI).

Reply:
Okay, in that case, ignore me and carry on!